ipaserver/install/server/replicainstall.py

```diff
@@ -508,22 +509,60 @@ def promote_openldap_conf(hostname, master):
         root_logger.info("Failed to update {}: {}".format(ldap_conf, e))
 
 
-def check_remote_version(api):
+@contextlib.contextmanager
+def rpc_client(api):
+    """
+    Context manager for JSON RPC client.
+
+    :param api: api to initiate the RPC client
+    """
     client = rpc.jsonclient(api)
     client.finalize()
-
     client.connect()
+
     try:
-        env = client.forward(u'env', u'version')['result']
+        yield client
     finally:
         client.disconnect()
 
+
+def check_remote_fips_mode(client, local_fips_mode):
+    """
+    Verify remote server's fips-mode is the same as this server's fips-mode
+
+    :param client: RPC client
+    :param local_fips_mode: boolean indicating whether FIPS mode is turned on
+
+    :raises: ``ScriptError`` if the checks fails
+    """
+    env = client.forward(u'env', u'fips_mode')['result']
+    remote_fips_mode = env.get('fips_mode', False)
+    fips_mode = tasks.is_fips_enabled()
+    if fips_mode != remote_fips_mode:
+        if fips_mode:
+            raise ScriptError(
+                "Cannot join FIPS-enabled replica into existing topology: "
+                "FIPS is not enabled on the master server.")
+        else:
+            raise ScriptError(
+                "Cannot join replica into existing FIPS-enabled topology: "
+                "FIPS has to be enabled locally first.")
+
+
+def check_remote_version(client, local_version):
+    """
+    Verify remote server's version is not higher than this server's version
+
+    :param client: RPC client
+
```

Comment:
Please fill this empty line with the other parameter documentation. Don't leave a blank line before `:raises`, it's ugly and it's not usually done this way [pythonhosted:Sphinx](https://pythonhosted.org/an_example_pypi_project/sphinx.html).